src/contracts/pods/EigenPod.sol

```diff
@@ -428,6 +450,104 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         _sendETH(recipient, amountWei);
     }
 
+    function requestPartialWithdrawalsProof(
+        uint64 oracleTimestamp,
+        uint64 endTimestamp,
+        address recipient,
+        uint32 callbackGasLimit
+    ) external payable onlyEigenPodOwner hasEnabledRestaking {
+        // initializes the  "timestampProvenUntil" for batched zk partial withdrawals to the GENESIS_TIME
+        if(timestampProvenUntil == 0){
+            timestampProvenUntil = GENESIS_TIME;
+        }
+
+        requestNonce++;
+        require(endTimestamp > timestampProvenUntil, "EigenPod.submitPartialWithdrawalsBatchForVerification: endTimestamp must be greater than timestampProvenUntil");
+        
+        //the proofs are made [startTimestamp, endTimestamp), so the next startTimestamp = previous endTimestamp     
+        uint64 startTimestamp = timestampProvenUntil;
+        
+        //record the partial withdrawal proof request
+       _partialWithdrawalProofRequests[requestNonce] = PartialWithdrawalProofRequest({
+            startTimestamp: startTimestamp,
+            endTimestamp: endTimestamp,
+            recipient: recipient,
```

Comment:
Unless I'm misunderstanding something, shouldn't `recipient` be `address(this)`? I'm pretty sure I've gotten something wrong lol